modules/system/classes/MixAssets.php

```diff
@@ -72,26 +83,69 @@ public function init()
                 }
 
                 foreach ($packageArray as $name => $package) {
-                    $this->registerPackage($name, PluginManager::instance()->getPluginPath($pluginCode), $package);
+                    $this->registerPackage($name, PluginManager::instance()->getPluginPath($pluginCode) . '/' . $package);
                 }
             }
         }
 
-        // Allow current theme to define mix assets
-        $theme = Theme::getActiveTheme();
+        // Get the currently enabled modules
+        $enabledModules = Config::get('cms.loadModules', []);
+
+        if (in_array('Cms', $enabledModules)) {
+            // Allow current theme to define mix assets
+            $theme = Theme::getActiveTheme();
+
+            if (!is_null($theme)) {
+                $mix = $theme->getConfigValue('mix', []);
 
-        if (!is_null($theme)) {
-            $mix = $theme->getConfigValue('mix', []);
+                if (count($mix)) {
+                    foreach ($mix as $name => $file) {
+                        $this->registerPackage($name, $theme->getPath() . '/' . $file);
+                    }
+                }
+            }
+        }
+
+        $packagePaths = [];
+
+        // Search modules for Mix packages to autoregister
+        foreach ($enabledModules as $module) {
+            $module = strtolower($module);
+            $path = base_path("modules/$module") . '/' . $this->mixJs;
+            if (File::exists($path)) {
+                $packagePaths["module-$module"] = $path;
+            }
+        }
 
-            if (count($mix)) {
-                foreach ($mix as $name => $file) {
-                    $path = PathResolver::resolve($theme->getPath() . '/' . $file);
-                    $pinfo = pathinfo($path);
+        // Search plugins for Mix packages to autoregister
+        $plugins = PluginManager::instance()->getPlugins();
+        foreach ($plugins as $plugin) {
+            $path = $plugin->getPluginPath() . '/' . $this->mixJs;
+            if (File::exists($path)) {
+                $packagePaths[$plugin->getPluginIdentifier()] = $path;
+            }
+        }
 
-                    $this->registerPackage($name, $pinfo['dirname'], $pinfo['basename']);
+        // Search themes for Mix packages to autoregister
+        if (in_array('Cms', $enabledModules)) {
+            $themes = Theme::all();
+            foreach ($themes as $theme) {
+                $path = $theme->getPath() . '/' . $this->mixJs;
+                if (File::exists($path)) {
+                    $packagePaths["theme-" . $theme->getId()] = $path;
```

Comment:
Do we want to allow unused / disabled themes to be built as well? I think for the most part, we'd only want to build the active theme.

Reply:
I don't see why not. Personally for our multisite usage of winter we'll have a theme for every site and yet winter considers only one of them "active" at a time. There's always the p filter option